src/Login/containers/LoginContainer.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Login from '../components/LoginComponent';
+import { loginUtil } from '../redux/validations';
+import loginAction from '../redux/actions/LoginAction';
+
+export class LoginContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      errors: {
+        invalid_email:false,
+      }
+    };
+  }
+  onChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+    const { errors } = this.state;
+    switch(name) {
+      case "email":
+        if (!this.checkValidEmail(value)) {
+          this.setState({ errors: { userEmail: 'Please provide a valid email address' } });
+          errors.invalid_email = true;
+        }else{
+          this.setState({ errors: { userEmail: '' } });
+          errors.invalid_email = false;
+        }
+        break;
+      default:
+        break;
+    }
+
+    this.setState({ [event.target.name]: event.target.value });
+
+  }
+
+  checkLoginStatus = () => {
+    const {isloggedIn} = this.props.login;
+    if(!isloggedIn){
+      this.setState({ is_form_submited: false});
+    }
+  }
+  checkValidEmail = (userEmail) => {
+    const mailFormat = loginUtil.LOGIN_UTILS;
+    if (userEmail.match(mailFormat)) {
+      return true;
+    }
+    return false;
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const {email, password} = this.state
+    const userdata = {
+      user: {
+        email,
+        password
+      }
+    };
+    
+    if (email === "" ) {
+      this.setState({ errors: { userEmail: 'Email field required' } })
+    }
+    if (password ==="" ) {
+      this.setState({ errors: { userPassword: 'Password field is required'} });
+    }else (!this.Setstate)
+     
+    if(!this.state.errors.invalid_email && this.state.errors.userEmail===''){
+      this.props.loginAction(userdata, this.props);
+    }
+  }
+
+  render() {
+    return (
+      <Login
+        handleSubmit={this.handleSubmit}
+        onChange={this.onChange}
+        email={this.state.email}
+        password={this.state.password}
+        errors={this.state.errors}
+        isLoggingIn={this.props.isLoggingIn}
+        loginFailed = {this.props.login_failed}
+        isLoading = {this.props.isLoading}
+      />
+    );
+  }
+}
+
+LoginContainer.propTypes = {
+  loginAction: PropTypes.func,
```

Comment:
I think this function prop is required `loginAction: PropTypes.func.isRequired`

Reply:
@AGMETEOR  kindly note l have worked on this as well thank you.